File: python/subscriptions/customers.py
import os
import time

from pagou.http import PagouHttpClient


# Customers are not exposed as a distinct SDK resource in any language, so the
# subscription example creates or reuses a customer through the raw client.
def create_or_reuse_customer(client: PagouHttpClient) -> dict:
    """Reuses PAGOU_CUSTOMER_ID when set, otherwise creates a fresh customer."""
    existing = os.environ.get("PAGOU_CUSTOMER_ID")
    if existing:
        return client.request_data("GET", f"/v2/customers/{existing}").data

    suffix = int(time.time() * 1000)
    payload = {
        "name": "Ana Souza",
        "email": f"ana.souza+{suffix}@example.com",
        "document": {"type": "CPF", "number": "19100000000"},
        "phone": "11999990000",
        "externalRef": f"cust_{suffix}",
    }
    return client.request_data("POST", "/v2/customers", body=payload).data

File: php/subscriptions/Customers.php
<?php

declare(strict_types=1);

namespace Pagou\Examples\Subscriptions;

use Pagou\Examples\HttpClient;

// Customers are not an SDK resource in any language; both the subscription
// examples create/reuse a customer through the raw HTTP client.
final class Customers
{
    /**
     * Reuses PAGOU_CUSTOMER_ID when set, otherwise creates a fresh customer.
     * @return array<string, mixed>
     */
    public static function createOrReuse(HttpClient $client): array
    {
        $existing = getenv('PAGOU_CUSTOMER_ID') ?: null;
        if ($existing) {
            $result = $client->requestData(['method' => 'GET', 'path' => "/v2/customers/{$existing}"]);
            return $result['data'];
        }

        $suffix = (int) (microtime(true) * 1000);
        $input = [
            'name' => 'Ana Souza',
            'email' => "ana.souza+{$suffix}@example.com",
            'document' => ['type' => 'CPF', 'number' => '19100000000'],
            'phone' => '11999990000',
            'externalRef' => "cust_{$suffix}",
        ];

        $result = $client->requestData(['method' => 'POST', 'path' => '/v2/customers', 'body' => $input]);
        return $result['data'];
    }
}
